spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTemplateTests.java

```diff
@@ -336,7 +355,53 @@ SendTestArgs producerCustomizer() {
 
 	}
 
-	record Foo(String foo, String bar) {
+	public static class Foo {
```

Comment:
AVRO doesn't work with records atm

Reply:
Yeh, it is unfortunate. You will see a `User` and `User2` scattered about in a few tests because of this.

Reply:
Doing this 2x now has me on the verge of bringing in Lombok just for testing and `@Data` in this case. The next time I find myself doing this, its Lombok time. :)

Reply:
Personal opinion: please refrain 😄 ! Using Lombok has a lot of implications (like imposing to use an IDE that supports it). So I would avoid it for as long as possible. There are not that many value objects and records can be used most of the time.

Reply:
Support for record in AVRO : https://github.com/apache/avro/pull/1680
Probably for version 1.12

Reply:
>  please refrain 😄 ! Using Lombok has a lot of implications 
Yep, just venting mostly. If we do end up bringing it in it would only be at test dep level. But yeh, 2 of these manual pojos does not warrant it.